compiler/psi/src/org/jetbrains/kotlin/parsing/KotlinExpressionParsing.java

```diff
@@ -1836,8 +1841,17 @@ public void parseValueArgumentList() {
                 while (true) {
                     while (at(COMMA)) errorAndAdvance("Expecting an argument");
                     parseValueArgument();
-                    if (at(COLON) && lookahead(1) == IDENTIFIER) {
-                        errorAndAdvance("Unexpected type specification", 2);
+                    if (at(COLON) && (lookahead(1) == IDENTIFIER || lookahead(1) == LPAR)) {
```

Comment:
BTW a type may not be started with `IDENTIFIER`. It can be a functional type, an annotated type or suspend functional type. See [the grammar](https://kotlinlang.org/docs/reference/grammar.html#types) for more details.

You can just try to parse a type after `:` and report `Unexpected type specification ` if it succeeded.

Reply:
It seems, it can be simplified (in this version):
```
if (at(COLON) && (lookahead(1) == IDENTIFIER || lookahead(1) == LPAR)) {
    PsiBuilder.Marker err = mark();
    advance(); // COLON
    myKotlinParsing.parseTypeRef();
    err.error("Unexpected type specification");
}
```
But see the comment above.